### DISCUSSION TITLE
Wazuh future

### DISCUSSION THREAD

**Discussion by itdeveloper50** [Security-Onion-Solutions/securityonion#9834] [Unsupported Versions]
On Security Onion documentation it is written:

`Before deploying lots of Wazuh agents to your enterprise, please be aware that Wazuh will not be included in the upcoming Security Onion 2.4.`

1. What is the reason of not being included of Wazuh agents in future releases?
2. What do you advise to use instead of Wazuh agent?
3. Will Wazuh agent be replaced with new type agent?

**Comment by adamphetamine** [comment]
I think there's a fair bit of overlap in capabilities with OSQuery /Fleet/Wazuh. In fact Fleet and Elastic Fleet both include OSQuery, so it makes sense to deprecate Fleet in favour of Elastic. 
I was also told that the integration had not been upgraded for some time because they discovered an issue or conflict. 
Sorry this isn't super detailed, but I hope it helps

**Comment by adamphetamine** [comment]
oops actually this has more detail-

https://blog.securityonion.net/2023/02/security-onion-in-2022-and-2023.html

'Since Elastic Agent has osquery built in, it will be taking the place of the current osquery agent. Security Onion 2.4 will also use the Elastic Agent to send alerts and metadata from the sensors to the back end, replacing the current Filebeat agent. Users will be able to manage all of their Elastic Agents using Elastic Fleet in Kibana. Since Elastic Agent covers most of the Wazuh use cases used in Security Onion, Wazuh is being removed as well. This single agent architecture will save resources, streamline administrative processes, and ease the upgrade process in Security Onion.'
